cmd/internal/applicationsnapshot/source.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applicationsnapshot
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/hashicorp/go-getter"
+)
+
+const SCHEMA = "https://raw.githubusercontent.com/redhat-appstudio/release-service/main/config/crd/bases/appstudio.redhat.com_applicationsnapshots.yaml"
+
+type fetcherFn func() (*os.File, error)
+
+type schemaSource struct {
+	fetch fetcherFn
+}
+
+// NewSchemaSource constructs the default schemaSource
+func NewSchemaSource() schemaSource {
+	return schemaSource{
+		fetch: fetch,
+	}
+}
+
+// fetch the latest CRD and return the file
+//go:generate build
+func fetch() (*os.File, error) {
+	// Create a tempfile for the CRD
+	file, err := ioutil.TempFile(os.TempDir(), "ec-cli-applicationsnapshot")
```

Comment:
For consistency of nitpickery

```suggestion
	file, err := ioutil.TempFile("", "ec-cli-applicationsnapshot")
```

Reply:
With `io.reader` we don't need a tmpfile anymore.